src/core/src/op/unsqueeze.cpp

```diff
@@ -33,120 +32,89 @@ void op::v0::Unsqueeze::validate_and_infer_types() {
     OV_OP_SCOPE(v0_Unsqueeze_clone_with_new_inputs);
     if (new_args.size() != 2) {
         OPENVINO_THROW("Incorrect number of new arguments");
     }
-    return make_shared<Unsqueeze>(new_args.at(0), new_args.at(1));
+    return std::make_shared<Unsqueeze>(new_args.at(0), new_args.at(1));
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
 namespace ov {
 namespace op {
 namespace unsqueeze {
-struct Evaluate : element::NoAction<bool> {
-    using element::NoAction<bool>::visit;
-
-    template <element::Type_t ET>
-    static result_type visit(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count) {
-        ov::reference::copy(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count);
-        return true;
-    }
-};
 
 // The evaluate cannot use shape_infer for output shape calculation as shape inference accepts
 // repeated axis and evaluate not. When shape inference will changed to be compatible with `numpy` then
 // evaluate and inference can use same function to calculate output shape. TODO for next version for this operator.
 namespace {
-bool evaluate_unsqueeze(const Node* node,
-                        const HostTensorPtr& arg0,
-                        const HostTensorPtr& arg1,
-                        const HostTensorPtr& out) {
-    auto element_type = arg0->get_element_type();
-    out->set_element_type(element_type);
-
-    const auto& axes_shape = arg1->get_shape();
+bool evaluate_unsqueeze(const Node* node, const ov::Tensor& arg0, const ov::Tensor& arg1, ov::Tensor& out) {
+    const auto& axes_shape = arg1.get_shape();
     ov::op::v0::check_unsqueeze_axes_rank(node, Rank(axes_shape.size()));
 
-    const auto& data_shape = arg0->get_shape();
+    const auto& data_shape = arg0.get_shape();
     const auto out_rank = static_cast<int64_t>(data_shape.size() + shape_size(axes_shape));
 
     // Get axes and normalize
+    auto axes = ov::util::read_index_vector(arg1);
     OPENVINO_SUPPRESS_DEPRECATED_START
-    auto axes = read_index_vector(arg1);
-    normalize_axes(node, out_rank, axes);
+    ov::normalize_axes(node, out_rank, axes);
     OPENVINO_SUPPRESS_DEPRECATED_END
 
     // Sort in increasing order
     std::set<int64_t> axes_set(axes.begin(), axes.end());
-    NGRAPH_CHECK(axes.size() == axes_set.size(), "Axes has duplicate axis.");
+    OPENVINO_ASSERT(axes.size() == axes_set.size(), "Axes has duplicate axis.");
 
     auto out_shape = data_shape;
     for (int64_t axis : axes_set) {
         out_shape.insert(out_shape.begin() + axis, 1);
     }
-    out->set_shape(out_shape);
-
-    using namespace ov::element;
-    return IfTypeOf<i32, i64, u32, u64, f16, f32, f64, bf16>::apply<Evaluate>(element_type,
-                                                                              arg0,
-                                                                              out,
-                                                                              shape_size(out_shape));
+    out.set_shape(out_shape);
```

Comment:
seems like we can reuse `shape_infer` to get `out_shape` and avoid duplicating shape inference logic. See Tile evaluate on how it could be done.

Reply:
Changed